### DISCUSSION TITLE
Data partitioning

### DISCUSSION THREAD

**Discussion by weralabaj** [RailsEventStore/rails_event_store#1163] [Ideas]
In some systems events and streams could be partitioned and saved in multiple tables. For example, in a multi-tenant system where users from various tenants don't interact with each other, the data could be partitioned by tenant id.

Potential benefits in such a scenario:
- more efficient queries/statements even when event store has a lot of data
- easy to move/copy data of a single tenant

**Comment by mostlyobvious** [comment]
Possibly related: https://www.honeybadger.io/blog/pg-partition-manager/
